host/src/main/java/org/area515/resinprinter/job/STLFileProcessor.java

```diff
@@ -124,6 +129,38 @@ public JobStatus processFile(PrintJob printJob) throws Exception {
 		}
 	}
 	//TODO: Create PreviewSlice0 method that copies processfile code
+	public void previewSlice(PrintJob printJob) throws Exception {
+		try {
+			//Initialize DataAid
+			//TODO: Create dataaid manually based on started printer
+			//how do access list of printers?

```

Comment:
Take a look at PrintableService.print() to see how to get a list of printers and determine if one is started, and also how to throw an exception that will be displayed in the UI if there are no started printers.